### DISCUSSION TITLE
Blogging with jupyterbook

### DISCUSSION THREAD

**Discussion by CalMacCQ** [executablebooks/meta#1009] [General] [answered]
Hi,

I have been toying with the idea of blogging with jupyterbook for a while.

The blogs would be written in markdown  and contain code cells as well interactive diagrams. 

I wanted the blog to match the theme of the API documentation I use in my project (I use the sphinx-book-theme). This made jupyter book a natural choice. I also like fact that users could easily download my notebook and play around with it.

I was wondering anyone else had tried using sphinx-book-theme or jupuyterbook for blogs and if there are any specific features I should be aware of. 

So far I've come accross this page ->  https://sphinx-book-theme.readthedocs.io/en/stable/reference/blog.html

**Comment by choldgraf** [comment, marked as answer]
If you're comfortable with Sphinx already you could always use ablog directly in a Sphinx site. For one example see my repository: 

https://github.com/choldgraf/choldgraf.github.io

Here's the blog folder with content

https://github.com/choldgraf/choldgraf.github.io/tree/main/blog

And here is the conf.py - all the variables starting with `ablog` are configuring the blog. See this section:

https://github.com/choldgraf/choldgraf.github.io/blob/aabaf19a91abe980e7581611b228c1fd6c17b5bf/conf.py#L107

You can also add ablog components to the sidebars like here:

https://github.com/choldgraf/choldgraf.github.io/blob/aabaf19a91abe980e7581611b228c1fd6c17b5bf/conf.py#L70

To do this in jupyter book you could basically add the same configuration manually in the config.yaml file if you like. Here's a starting point for that:

https://jupyterbook.org/en/stable/advanced/sphinx.html

**Reply by CalMacCQ** [reply]
Thank you for your thorough response. That is very helpful.
